### PR TITLE
[MRRTF-210] Migrate MCH cluster I/O to IO and DevIO libs

### PR DESCRIPTION
**Pull request by aphecetche**
Step 3 of 4 of the migration to IO lib.

### CODE REVIEW

File: Detectors/MUON/MCH/Workflow/CMakeLists.txt [hunk 1]
```diff
@@ -122,30 +96,6 @@ o2_add_executable(
...
-        SOURCES src/tracks-reader-workflow.cxx
-        COMPONENT_NAME mch
-        PUBLIC_LINK_LIBRARIES O2::MCHWorkflow)
-
-o2_add_executable(
-        tracks-sampler-workflow
-        SOURCES src/TrackSamplerSpec.cxx src/tracks-sampler-workflow.cxx
-        COMPONENT_NAME mch
-        PUBLIC_LINK_LIBRARIES O2::Framework O2::DataFormatsMCH O2::MCHBase)
-
```
Comment (pillot) [inline Detectors/MUON/MCH/Workflow/CMakeLists.txt]: you are a step ahead in the removal of the track IO workflows ;-)

File: Detectors/MUON/MCH/Workflow/CMakeLists.txt [hunk 2]
```diff
@@ -187,15 +137,11 @@ o2_add_executable(
...
+            O2::MCHIO
             O2::SimulationDataFormat
             O2::Steer
         TARGETVARNAME mch-reco-workflow)
 
-o2_add_executable(tracks-file-dumper
-        SOURCES src/tracks-file-dumper.cxx
-        COMPONENT_NAME mch
-        PUBLIC_LINK_LIBRARIES Boost::program_options O2::MCHWorkflow)
-
```
Comment (pillot) [inline Detectors/MUON/MCH/Workflow/CMakeLists.txt]: same here

File: Detectors/MUON/MCH/Workflow/src/ClusterFinderGEMSpec.cxx [hunk 3]
```diff
@@ -282,9 +282,9 @@ class ClusterFinderGEMTask
 
       // fill the ouput messages
       if (isActive(GEMOutputStream)) {
-        clusterROFs.emplace_back(preClusterROF.getBCData(), clusters.size(), mClusterFinderGEM.getClusters().size());
+        clusterROFs.emplace_back(preClusterROF.getBCData(), clusters.size(), mClusterFinderGEM.getClusters().size(), preClusterROF.getBCWidth());
       } else {
-        clusterROFs.emplace_back(preClusterROF.getBCData(), clusters.size(), mClusterFinderOriginal.getClusters().size());
+        clusterROFs.emplace_back(preClusterROF.getBCData(), clusters.size(), mClusterFinderOriginal.getClusters().size(), preClusterROF.getBCWidth());
       }
```
Comment (pillot) [inline Detectors/MUON/MCH/Workflow/src/ClusterFinderGEMSpec.cxx]: Are these changes intended in this PR?

File: Detectors/MUON/MCH/Workflow/CMakeLists.txt (hunk 1 above)
Comment (aphecetche) [reply Detectors/MUON/MCH/Workflow/CMakeLists.txt]: indeed ;-) fixed.

File: Detectors/MUON/MCH/Workflow/src/ClusterFinderGEMSpec.cxx (hunk 3 above)
Comment (aphecetche) [reply Detectors/MUON/MCH/Workflow/src/ClusterFinderGEMSpec.cxx]: nope. good catch. fixed.

File: Detectors/MUON/MCH/Workflow/CMakeLists.txt (hunk 1 above)
Comment (pillot) [reply Detectors/MUON/MCH/Workflow/CMakeLists.txt]: not completely. You only put back 2 out of 5 track related workflows.

Comment (aphecetche) [reply Detectors/MUON/MCH/Workflow/CMakeLists.txt]: ok, let me try again...

**Review by pillot — APPROVED** [review]

**Comment by aphecetche** [comment]
@shahor02 just to let you know that this is another PR where we'd need your approval (touching 2 GlobalTrackingWorkflow CMakeLists.txt). Thanks !

**Review by shahor02 — APPROVED** [review]
